test/e2e/storage/testsuites/provisioning.go

```diff
@@ -305,36 +305,76 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 					}
 					e2evolume.TestVolumeClientSlow(f, myTestConfig, nil, "", tests)
 				}
-				myTestCase.TestDynamicProvisioning()
+				t.TestDynamicProvisioning()
 			}(i)
 		}
 		wg.Wait()
 	})
 }
 
+// ProvisionStorageClass provisions a StorageClass from a spec, if the StorageClass already exists
+// then it returns it as a `computed` StorageClass, if it doesn't exist then it's created first
+// and then returned, if the spec is nil then we return the `default` StorageClass
+func ProvisionStorageClass(
+	client clientset.Interface,
+	claim *v1.PersistentVolumeClaim,
+	class *storagev1.StorageClass,
+) (*storagev1.StorageClass, func()) {
+	gomega.Expect(client).NotTo(gomega.BeNil(), "ProvisionStorageClass.client is required")
+	gomega.Expect(claim).NotTo(gomega.BeNil(), "ProvisionStorageClass.claim is required")
+
+	var err error
+	var computedStorageClass *storagev1.StorageClass
+	var clearComputedStorageClass = func() {}
+	if class != nil {
+		framework.ExpectEqual(*claim.Spec.StorageClassName, class.Name)
+
+		computedStorageClass, err = client.StorageV1().StorageClasses().Get(context.TODO(), class.Name, metav1.GetOptions{})
+		if err == nil {
+			// skip storageclass creation if it already exists
+			ginkgo.By("Storage class " + computedStorageClass.Name + " is already created, skipping creation.")
+		} else {
+			ginkgo.By("Creating a StorageClass " + class.Name)
+			_, err = client.StorageV1().StorageClasses().Create(context.TODO(), class, metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+			computedStorageClass, err = client.StorageV1().StorageClasses().Get(context.TODO(), class.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			clearComputedStorageClass = func() {
+				framework.Logf("deleting storage class %s", computedStorageClass.Name)
+				framework.ExpectNoError(client.StorageV1().StorageClasses().Delete(context.TODO(), computedStorageClass.Name, metav1.DeleteOptions{}))
+			}
+		}
+	} else {
+		// StorageClass is nil, so the default one will be used
+		scName, err := e2epv.GetDefaultStorageClassName(client)
+		framework.ExpectNoError(err)
+		computedStorageClass, err = client.StorageV1().StorageClasses().Get(context.TODO(), scName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		// If late binding is configured, create and delete a pod to provision the volume
+		if *computedStorageClass.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+			ginkgo.By("creating a pod referring to the claim")
+			var pod *v1.Pod
+			pod, err := e2epod.CreatePod(client, claim.Namespace, nil /* nodeSelector */, []*v1.PersistentVolumeClaim{claim}, true /* isPrivileged */, "" /* command */)
+			// Delete pod now, otherwise PV can't be deleted below
+			framework.ExpectNoError(err)
+			e2epod.DeletePodOrFail(client, pod.Namespace, pod.Name)
+		}
+	}
+
+	return computedStorageClass, clearComputedStorageClass
+}
+
 // TestDynamicProvisioning tests dynamic provisioning with specified StorageClassTest
```

Comment:
Can you add a comment here that this method expects the StorageClass to already exist.

Reply:
Got it